components/EmailAuthModal.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from "react";
+import {
+  Button,
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  Input,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useToast,
+} from "@chakra-ui/react";
+import { useSessionContext } from "@supabase/auth-helpers-react";
+
+export const EmailAuthModal = ({
+  isOpen,
+  onClose,
+  onOpen,
+}: {
+  isOpen: boolean;
+  onClose: () => void;
+  onOpen: () => void;
+}) => {
+  const [loading, setLoading] = React.useState(false);
+  const [email, setEmail] = React.useState("");
+  const toast = useToast();
+
+  const { supabaseClient } = useSessionContext();
+
+  const initialRef = React.useRef(null);
+
+  const onSendLoginLink = async () => {
+    try {
+      setLoading(true);
+      const { error } = await supabaseClient.auth.signInWithOtp({
```

Comment:
pretty dope - this works for both new and existing users. both use cases are tested in the vid provided - you'll notice I got two different emails